controllers/managedocs_controller_test.go

```diff
@@ -839,6 +860,60 @@ var _ = Describe("ManagedOCS controller", func() {
 				utils.WaitForResource(k8sClient, ctx, amConfigSecretTemplate.DeepCopy(), timeout, interval)
 			})
 		})
+		When("there is grafana-datasources secret in openshift-monitoring namespace", func() {
+			It("should create k8s-metrics-service-monitor-auth secret in primary namespace", func() {
+				// Create grafana-datasources secret
+				authInfoStructure := struct {
+					DataSources []struct {
+						BasicAuthPassword string `json:"basicAuthPassword"`
+						BasicAuthUser     string `json:"basicAuthUser"`
+					} `json:"datasources"`
+				}{}
+				authInfoStructure.DataSources = make([]struct {
+					BasicAuthPassword string "json:\"basicAuthPassword\""
+					BasicAuthUser     string "json:\"basicAuthUser\""
+				}, 1)
+				authInfoStructure.DataSources[0].BasicAuthPassword = "password"
+				authInfoStructure.DataSources[0].BasicAuthUser = "internal"
+				grafanaSecret := grafanaDatasourceSecretTemplate.DeepCopy()
+				grafanaSecret.Data = make(map[string][]byte)
+				grafanaSecret.Data["prometheus.yaml"], _ = json.Marshal(authInfoStructure)
+				Expect(k8sClient.Create(ctx, grafanaSecret)).Should(Succeed())
+				// Check for federate-basic-auth secret in primary namespace
+				utils.WaitForResource(k8sClient, ctx, testK8sMetricsServiceMonitorAuthSecretTemplate.DeepCopy(), timeout, interval)
+			})
+		})
+		When("there is k8s-metrics-service-monitor-auth secret in primary namespace", func() {
+			It("should create k8s-metrics-service-monitor service monitor", func() {
+				secret := testK8sMetricsServiceMonitorAuthSecretTemplate.DeepCopy()
+				secretKey := utils.GetResourceKey(secret)
+				Expect(k8sClient.Get(ctx, secretKey, secret)).Should(Succeed())
+				sm := testK8sMetricsServiceMonitorAuthSecretTemplate.DeepCopy()
+				smKey := utils.GetResourceKey(sm)
+				Expect(k8sClient.Get(ctx, smKey, sm)).Should(Succeed())
+			})
+		})
```

Comment:
This case is irrelevant, the service monitor should be created in any case. 
They are non-coupled (but ordered) operations.